### DISCUSSION TITLE
starkandwayne hello-world tutorial: failed to ping registry

### DISCUSSION THREAD

**Discussion by ben-kenney** [concourse/concourse#7161] [Help & Support]
Hi all, I'm just getting into concourse and I'm stuck with the most simple hello-world tutorial case. 
I'm following the [Stark and Wayne tutorial](https://concoursetutorial.com/basics/task-hello-world/), setting concourse up in docker (which is pulling v7.3.1). 

Everything seems fine but it hangs/fails when executing task_hello_world.yml. It can't download the busybox container.

This is the output:

> executing build 2 at http://127.0.0.1:8080/builds/2
initializing
selected worker: b35a9aa6f14a
failed to ping registry: 2 error(s) occurred:
>
> ping https: Get "https://registry-1.docker.io/v2/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
> ping http: Get "http://registry-1.docker.io/v2/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
image check failed
errored

Indeed I cannot ping https://registry-1.docker.io, any tips on resolving this?

At first I thought perhaps it was a proxy setting preventing access to the docker.io site, but I've tried on a few servers and can't get this to work.

I followed the tutorial exactly (running through docker-compose), I can access the web interface and login without any issues.

Any help would be appreciated!

**Comment by ben-kenney** [comment]
So it turns out that this must be a docker problem, not a concourse problem. When I manually edit the /etc/resolv.conf file in the concourse docker container to add "nameserver 8.8.8.8" then I am able to get the hello-world tutorial to work.

**Reply by aoldershaw** [reply]
You can also set `CONCOURSE_GARDEN_DNS_PROXY_ENABLE: true` in your `docker-compose.yml` - that tutorial should probably be updated accordingly

**Reply by aoldershaw** [reply]
Actually, looks like the tutorial uses the `quickstart` command - in which case you should set `CONCOURSE_WORKER_GARDEN_DNS_PROXY_ENABLE: true`

**Reply by IAXES** [reply]
I can put the patch in place easily enough (example below). However, is this the one-size-fits all solution that should be the new default, or should this instead be part of some `README.md` that notes "if issue XYZ arises, he's the fix" (i.e. apply this patch).

```diff
index df1dd2d..b7f9f63 100644
--- a/docker-compose.yml
+++ b/t.yml
@@ -24,4 +24,3 @@ services:
       - CONCOURSE_EXTERNAL_URL
       - CONCOURSE_ADD_LOCAL_USER=admin:admin
       - CONCOURSE_MAIN_TEAM_LOCAL_USER=admin
-      - CONCOURSE_WORKER_GARDEN_DNS_PROXY_ENABLE=true
```

**Reply by IAXES** [reply]
Cross-linking with related issue (already closed): https://github.com/concourse/concourse/issues/2388
